commit 38ab2af2b57bc9390bb9b695001cba63b9262f2b
Author: Robert Lougher <rob.lougher@gmail.com>
Date:   Wed Dec 14 16:14:17 2016 +0000

    New API for merging debug locations. NFC.
    
    Given two debug locations the function getMergedLocation combines the
    locations into a single location (which may be an empty location).
    Please see https://reviews.llvm.org/D26256 for the discussion leading
    up to this API.
    
    Note the function is currently a stub.  This allows optimisations to
    use the API although no location will actually be used.
    
    This is patch 1 out of 8 for D26256.  As suggested by David Blaikie,
    each change in D26256 has been broken out into a separate patch.
    
    
    git-svn-id: https://llvm.org/svn/llvm-project/llvm/trunk@289661 91177308-0d34-0410-b5e6-96231b3b80d8

diff --git a/include/llvm/IR/DebugInfoMetadata.h b/include/llvm/IR/DebugInfoMetadata.h
index bda66ae9952..aec6601efc4 100644
--- a/include/llvm/IR/DebugInfoMetadata.h
+++ b/include/llvm/IR/DebugInfoMetadata.h
@@ -1291,6 +1291,24 @@ public:
   /// Returns a new DILocation with updated \p Discriminator.
   inline DILocation *cloneWithDiscriminator(unsigned Discriminator) const;
 
+  /// When two instructions are combined into a single instruction we also
+  /// need to combine the original locations into a single location.
+  ///
+  /// When the locations are the same we can use either location. When they
+  /// differ, we need a third location which is distinct from either. If
+  /// they have the same file/line but have a different discriminator we
+  /// could create a location with a new discriminator. If they are from
+  /// different files/lines the location is ambiguous and can't be
+  /// represented in a single line entry.  In this case, no location
+  /// should be set.
+  ///
+  /// Currently this function is simply a stub, and no location will be
+  /// used for all cases.
+  static DILocation *getMergedLocation(const DILocation *LocA,
+                                       const DILocation *LocB) {
+    return nullptr;
+  }
+
   Metadata *getRawScope() const { return getOperand(0); }
   Metadata *getRawInlinedAt() const {
     if (getNumOperands() == 2)
